shell_integration/windows/WinShellExt.wxs.in

```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ *
+ * Copyright (C) by Michael Schuster <michael.schuster@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+-->
+<?include $(var.ProjectPath)Platform.wxi?>
+
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+    <Fragment>
+
+        <!-- Context Menu -->
+        <?define ContextMenuGuid        = "@WIN_SHELLEXT_CONTEXT_MENU_GUID@" ?>
+        <?define ContextMenuRegKeyName  = "@APPLICATION_SHORTNAME@ContextMenuHandler" ?>
+
+        <?define ContextMenuDescription = "@APPLICATION_SHORTNAME@ context menu handler" ?>
+
+        <!-- Overlays -->
+        <?define OverlayGuidError       = "@WIN_SHELLEXT_OVERLAY_GUID_ERROR@" ?>
+        <?define OverlayGuidOK          = "@WIN_SHELLEXT_OVERLAY_GUID_OK@" ?>
+        <?define OverlayGuidOKShared    = "@WIN_SHELLEXT_OVERLAY_GUID_OK_SHARED@" ?>
+        <?define OverlayGuidSync        = "@WIN_SHELLEXT_OVERLAY_GUID_SYNC@" ?>
+        <?define OverlayGuidWarning     = "@WIN_SHELLEXT_OVERLAY_GUID_WARNING@" ?>
+
+        <!-- Preceeding spaces are intended, two spaces to put us ahead of the competition :/ -->
```

Comment:
??

OK... intended, and I like the joke in there, but maybe that warrant explaining the reason in the comment?

Reply:
This is taken over from the original implementation:
https://github.com/nextcloud/desktop/blob/cd2057f/shell_integration/windows/OCOverlays/OverlayConstants.h#L26-L29

Everyone working on this will get it latest, when you open `regedit` and look at the other keys there ;p
https://github.com/nextcloud/desktop/blob/cd2057f/shell_integration/windows/OCOverlays/OverlayConstants.h#L37

There is a limit in Windows (oh wonder^^) so that only the first 15 extensions get invoked, this is why they used that dirty little trick to get ahead ;)

I'll add a hint for this in the comment then.

Reply:
> This is taken over from the original implementation:
> https://github.com/nextcloud/desktop/blob/cd2057f/shell_integration/windows/OCOverlays/OverlayConstants.h#L26-L29
> 
> Everyone working on this will get it latest, when you open `regedit` and look at the other keys there ;p
> https://github.com/nextcloud/desktop/blob/cd2057f/shell_integration/windows/OCOverlays/OverlayConstants.h#L37
> 
> There is a limit in Windows (oh wonder^^) so that only the first 15 extensions get invoked, this is why they used that dirty little trick to get ahead ;)

... until they change the sort order for some reason. ;-)

But OK, I understand now, thanks for the information.

> I'll add a hint for this in the comment then.

Thanks!

Reply:
> 
> ... until they change the sort order for some reason. ;-)
> 

Haha, I highly doubt this 😂 With the Win32 innovation cycle.. 😉

Although I would be happy if they would provide such a change for us in the future :)

> But OK, I understand now, thanks for the information.

You're welcome ;)